lib/event/member_joined_channel.rb

```diff
@@ -0,0 +1,42 @@
+require 'operationcode/slack'
+
+# Handler for a Member Joined Channel event. When a user joins a Slack channel, we want to:
+# * Check what channel is joined, and take additional actions on a per- channel basis as needed
+
+class Event
+  class MemberJoinedChannel < Event
+    attr_reader :user
+
+    def initialize(data, token: nil, logger: nil)
+      super
+      @channel = data['event']['channel']
+
+    # May need better way to conditionally assign template, if we have other channels that need welcome messages
+      if @channel == Event::MENTORS_INTERNAL_CHANNEL
+        @template = File.read(template_path + 'mentor_welcome_message.txt.erb')
+      end
+    end
+
+    def process
+      @user = Operationcode::Slack::User.new(@data['event']['user'])
+      log "Production mode: #{production_mode?}"
+      log "Welcoming mentor #{resolve_user_name}"
+
+      if @channel == Event::MENTORS_INTERNAL_CHANNEL
```

Comment:
If we get to this point in the code, and the channel does not meet this conditional, we will have already logged twice, but will not be sending the slack message.

Perhaps instead of the conditional being here, add a guard class as the first line in this method, on line 21, that ejects from the method if this is not the desired channel.  That way we do not log twice, with no follow up message. 

For example:

```ruby
def process
  return unless @channel == Event::MENTORS_INTERNAL_CHANNEL

  @user = Operationcode::Slack::User.new(@data['event']['user'])

  log "Production mode: #{production_mode?}"
  log "Welcoming mentor #{resolve_user_name}"

  welcome_mentor!
end
```  

On second thought, you can remove this guard clause, **and** the `@channel == Event::MENTORS_INTERNAL_CHANNEL` on line 15, and move that logic into the method that will need to be created in the `operationcode_bot.rb` file.

Then in that new method, you can do something like:

```
if it is the right channel
  call the new MemberJoinedChannel#process method
end
```

Reply:
@hpjaj That sounds like a good path to follow. The recent updates you added to the README shed some light on how `operationcode_bot.rb` ties into all of this.  I'll go over my first changeset and modify some things, most likely this weekend. Thanks for the eyes-on.